app/src/main/java/nerd/tuxmobil/fahrplan/congress/changes/ChangeListViewModelFactory.kt

```diff
@@ -0,0 +1,24 @@
+package nerd.tuxmobil.fahrplan.congress.changes
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import nerd.tuxmobil.fahrplan.congress.repositories.AppExecutionContext
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
+
+class ChangeListViewModelFactory(
+
+    private val appRepository: AppRepository
+
+) : ViewModelProvider.Factory {
+
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        val logging = Logging.get()
+        @Suppress("UNCHECKED_CAST")
+        return ChangeListViewModel(
+            repository = appRepository,
+            executionContext = AppExecutionContext,
+            logging = logging
+        ) as T
```

Comment:
Just wondering if a `try-catch` would be reasonable or even to use a more concrete type to avoid Suppress.

Reply:
I have seen and used this as the common pattern for the ViewModel factory. I wouldn't invest to much love here since it only exists to connect the pieces.